### PR TITLE
Fix issue in iface_driver error test

### PR DESCRIPTION
**Pull request by yalzhang**
In the error test, we should unload the vhost-net module and remove the
/dev/vhost_net character device as well. And the device will be
recovered with "modprobe vhost-net". So don't need to modify the driver
module in kernel.

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
run locally successfully
$ avocado run --vt-type libvirt virtual_network.iface_options.iface_driver.driver_vhost
JOB ID     : fa69dc08665f022458ba31481c11d8fc4d055daa
JOB LOG    : /root/avocado/job-results/job-2018-05-02T20.45-fa69dc0/job.log
 (1/3) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_vhost.error_test: PASS (35.05 s)
 (2/3) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_vhost.with_vhost_net: PASS (59.06 s)
 (3/3) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_vhost.without_vhost_net: PASS (58.90 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 173.18 s

File: libvirt/tests/src/virtual_network/iface_options.py [hunk 1]
```diff
@@ -484,18 +484,12 @@ def check_mcast_network(session):
...
-                driver_path = ("/lib/modules/%s/kernel/drivers/vhost/"
-                               "vhost_net.ko" % kvm_version)
-                driver_backup = driver_path + ".bak"
-                cmd = ("modprobe -r {0}; lsmod | "
-                       "grep {0}".format("vhost_net"))
-                if not process.system(cmd, ignore_status=True, shell=True):
+                # remove vhost driver on host and
+                # the character file /dev/vhost-net
+                cmd = ("modprobe -r {0}; "
+                       "rm -f /dev/vhost-net".format("vhost_net"))
```
Comment (chunfuwen) [inline libvirt/tests/src/virtual_network/iface_options.py]: "rm -f /dev/vhost-net" to "rm -f {0}"

Comment (yalzhang) [reply libvirt/tests/src/virtual_network/iface_options.py]: one is "vhost_net" ---> {0}, the other is "/dev/vhost-net", they are different.

Comment (chunfuwen) [reply libvirt/tests/src/virtual_network/iface_options.py]: so maybe rm -f /dev/{0}

**Pull request merged**